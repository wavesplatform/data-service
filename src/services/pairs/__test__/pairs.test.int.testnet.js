const { createPgDriver } = require('../../../db');
const loadConfig = require('../../../loadConfig');
const options = loadConfig();
const create = require('../index');

const amountAsset = 'CVRciuSiK8xiNJSRitAG9dGqcmfFPHvn9bcXtntnpuvp';
const priceAsset = 'WAVES';

describe('Pairs', () => {
  const service = create({
    drivers: { pg: createPgDriver(options) },
    emitEvent: () => () => null,
  });

  describe('get one pair', () => {
    it('should return Pair for one correctly', done => {
      service
        .get({
          amountAsset,
          priceAsset,
        })
        .run()
        .listen({
          onResolved: pair => {
            expect(pair).toMatchSnapshot();
            done();
          },
        });
    });

    it('should return null for non existing pair', done => {
      service
        .get({
          amountAsset: '111',
          priceAsset: '222',
        })
        .run()
        .listen({
          onResolved: nullable => {
            expect(nullable).toEqual(null);
            done();
          },
        });
    });
  });

  describe('get many pairs', () => {
    it('should return Pairs for one correctly', done => {
      service
        .mget([
          {
            amountAsset,
            priceAsset,
          },
        ])
        .run()
        .listen({
          onResolved: pairs => {
            expect(pairs).toMatchSnapshot();
            done();
          },
        });
    });

    it('should return null for non existing pairs', done => {
      service
        .mget([
          {
            amountAsset: '111',
            priceAsset: '222',
          },
        ])
        .run()
        .listen({
          onResolved: pairs => {
            expect(pairs.__type).toEqual('list');
            expect(pairs.data).toEqual([null]);
            done();
          },
        });
    });
  });
});
