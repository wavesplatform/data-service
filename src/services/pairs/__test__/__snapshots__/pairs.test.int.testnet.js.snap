// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pairs get many pairs should return Pairs for one correctly 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "pair",
      "data": Object {
        "firstPrice": "1e-7",
        "lastPrice": "1e-7",
        "volume": "0.1",
        "volumeWaves": "1e-8",
      },
    },
  ],
}
`;

exports[`Pairs get one pair should return Pair for one correctly 1`] = `
Object {
  "__type": "pair",
  "data": Object {
    "firstPrice": "1e-7",
    "lastPrice": "1e-7",
    "volume": "0.1",
    "volumeWaves": "1e-8",
  },
}
`;
